packages/core/src/StateNode.ts

```diff
@@ -193,6 +193,16 @@ export class StateNode<
         )
       : EMPTY_OBJECT) as StateNodesConfig<TContext, TEvent, TStateSchema>;
 
+    if (this.type === 'compound' && !this.config.initial) {
```

Comment:
just a general q: what's your stance regarding dev-only **errors**? Would it be maybe worth it to remove this branch of code in production?

Reply:
Errors are something I'd love to have extracted instead, using a similar solution to: https://github.com/formium/tsdx#error-extraction